dashboard/app/controllers/followers_controller.rb

```diff
@@ -14,6 +14,24 @@ def create
     redirect_to redirect_url, notice: I18n.t('follower.added_teacher', name: @section.teacher.name)
   end
 
+  # As a logged-in student, join a section.  Returns JSON.  Designed to be called by ajax request.
+  def create_async
+    @section = Section.find_by_code params[:section_code]
+    if @section
+      @section.add_student current_user
+      render json: {
+        result: "success",
+        sections: current_user.sections_as_student.map(&:summarize),
+        notice: I18n.t('follower.added_teacher', name: @section.teacher.name)
+      }
+    else
+      render json: {
+        result: "error",
+        sections: current_user.sections_as_student.map(&:summarize)
+      }
```

Comment:
It's interesting that your error cases return a 200, an `"error"` string, and a section list.  I think (and dashboard devs please correct me here) that we'd rather our APIs returned an HTTP error code - maybe a 404 in this case, since the requested section doesn't exist - and the sections list is redundant, since it should be unchanged in any error case.